### PR TITLE
With authserver

### PR DESCRIPTION
**Pull request by mohsenari**
- added authserver integration using init-discord-bot
- added differentiation between a URL containing the work "redirect" and a URL that redirects to another location
- fixed a bug to only send a get request when the URL has https:// or http://

### CODE REVIEW

**Comment by dylhack** [comment]
> cmd/prevent-link-redirection/prevent-link-redirection 

remove accidental binary file

File: internal/events.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			resp, err := http.Get(match[0])
+			if err != nil {
+				log.Fatalf("http.Get => %v", err.Error())
+			}
+			finalURL := resp.Request.URL.String()
+			msgAuthor:= msg.Author.ID
+			botMessage := ""
+			// Check if URL has "redirect" in it
+			if strings.Contains(match[0], "redirect") {
+				botMessage = "sent this <" + match[0] + "> which contains redirect";
```
Comment (dylhack) [inline internal/events.go]: I recommend looking into [Sprintf](https://golang.org/pkg/fmt/#Sprintf)

File: internal/events.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			// Check if the response URL is the same as request url
+			} else if finalURL != match[0] {
+				botMessage = "sent this <" + match[0] + "> which redirects to <" + finalURL + ">";
```
Comment (dylhack) [inline internal/events.go]: Samething as above

File: internal/bot.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	register, err := authClient.Register(
+		auth.Feature{
+			Name:        "Stop Redirect", // Give this bot / feature a name
+			Description: "Looks for links to check if they are redirecting to another web site.", // Describe what this bot is doing
+			Commands: []auth.SubCommand{ // list all the commands this bot / feature has
+				{
+					Name:        "help",           // Command name like "add"
+					Description: "This command describes what the bot does.",           // Describe what the command does
+					Example:     []string{""}, // [command name, argument 1, argument 2] like [add, #channel, #channel]
+				},
```
Comment (dylhack) [inline internal/bot.go]: Remove default comments like "Give this bot / feature a name"

**Review by dylhack — COMMENTED** [review]
Looks good, just a couple things.

File: internal/bot.go (hunk 3 above)
Comment (mohsenari) [reply internal/bot.go]: done

File: internal/events.go (hunk 2 above)
Comment (mohsenari) [reply internal/events.go]: used Sprintf instead of generic string concatenation

File: internal/events.go (hunk 1 above)
Comment (mohsenari) [reply internal/events.go]: used sprintf instead of generic string concatenation

**Comment by mohsenari** [comment]
> Looks good, just a couple things.

@dhghf done the changes. Thanks for review comments. Feel free to merge if it looks good.

**Review by dylhack — APPROVED** [review]

**Pull request merged**